### PR TITLE
chore(build): adds gulp-ng-annotate to build process

### LINKED ISSUE #124: Add angular annotations (add support for ng-strict-di)
Annotations may have to be added to fix problems when ng-strict-di is used.

### PR DESCRIPTION
**Pull request by PascalPrecht**
Ensures that generated distribution source files use safe
dependency injection annotation syntax.

Closes #124

### CODE REVIEW

**Comment by PascalPrecht** [comment]
@btford ping me if dist generation shouldn't be part of this commit.

File: dist/router.es5.js [hunk 1]
```diff
@@ -148,6 +148,7 @@ function routerViewPortDirective($animate, $compile, $controller, $templateReque
     }, viewPortName);
   }
 }
+routerViewPortDirective.$inject = ["$animate", "$compile", "$controller", "$templateRequest", "$rootScope", "$location", "$componentLoader", "$router"];
```
Comment (gkalpak) [inline dist/router.es5.js]: I haven't really looked into the new router (yet), so I probably miss something, but why are those `$inject`s necessary if we are passing it through `ngAnnotate` ?

**Comment by gkalpak** [comment]
As a reminder, if/when #122 gets merged, then `...ViewPort...` needs to be renamed to `...Viewport...` in this PR as well.

<sub>(If this PR gets merged before #122, then #122 needs to be updated.)</sub>

**Comment by PascalPrecht** [comment]
@gkalpak this is going to be autogenerated once this PR lands, so we don't have to take care of this manually :)

Good point anyway!

File: dist/router.es5.js (hunk 1 above)
Comment (PascalPrecht) [inline dist/router.es5.js]: Generated by ng-annotate. This is the only possible annotation since the source code uses named functions.

Comment (gkalpak) [inline dist/router.es5.js]: Sorry, I didn't notice it was in the `dist/` folder :)

**Comment by PascalPrecht** [comment]
This landed as https://github.com/angular/router/commit/255bc228df951773a5263f391f9aa4df6455e451

**Pull request closed without merging**